### ISSUE TITLE
Base URL Option for use with sub directory Reverse Proxy

### ISSUE THREAD

**Issue by arbitrarypunter** [JasonHHouse/gaps#193]
I've had a read through the docs and the open and closed issues and didn't spot anything related to this.

Would it be possible to add a config option for base URL for use with sub directory based reverse proxy setups?
The sort of thing would be a config option where i could specify it as for example /gaps, then when i setup a reverse proxy of <mydomain>/gaps all of the URL's would pass through the reverse proxy nicely and be handled by the gaps web server.

At the moment if i try to do this the page only partially loads as lots of the links are broken as they point at /<resource> rather than /gaps/<resource>

Thanks for your work on gaps, its been a great addition to my setup!

**Comment by JasonHHouse** [comment]
Yeah, I think I can put that together. So the idea is that index is at /gaps and everything else extends off that?

**Comment by arbitrarypunter** [comment]
That's exactly it.
At the moment the pages won't load correctly through the reverse proxy as the url's in the index page are relative to / rather than /gaps (or whatever URL base could be set to).

**Comment by JasonHHouse** [comment]
That makes sense. Let me look into it. It may be as simple as using an out of the box Spring Boot feature or I can just set them to /gaps

**Comment by JasonHHouse** [comment]
Thinking about it more. I think I'd need to make the call. More in line with how plex is configured currently. I also tend to use PUT for saving and test as PUT is idempotent.

`{
    address : {ip},
    port : {port},
    apiKey : {apiKey}
}`

**Comment by cxtal** [comment]
+1 Very much needed for reverse proxying - right now only accessible via IP and Port without Organizr and easy to forget the port given different services.

**Comment by JasonHHouse** [comment]
I made a fix for this. If you want to test it out before I merge you can pull alpha. If you're using docker you need to set the property BASE_URL=/gaps. If you're using Unraid, the template has been updated but you can add a new variable setting the key and value to BASE_URL and /gaps to test. 

`docker push housewrecker/gaps:alpha`

**Comment by cxtal** [comment]
Thanks for the update. Did as instructed but it is still not working properly.

nginx configuration:
```
       location /gaps {
               proxy_pass http://127.0.0.1:8484;
```

I noticed the following:

- All the top-level links are rendered properly ("Libraries", "Missing", "RSS", "Settings", etc...)
- All the other URLs are not and point to ```/``` instead of ```/gaps``` (from ```BASE_URL=/gaps```), for example, the ```js/libraries/jquery``` etc...

TL;DR the main links work but the page appears broken

Thanks again for your work; a lot of people looking to integrate stuff in their stack and gaps is a nice feature to have!

**Comment by JasonHHouse** [comment]
That's a bummer. I know what the issue is. I'll see what I can do to get that fixed for you and thanks!

**Comment by YouveGotMeowxy** [comment]
Does this work yet? I have added this to my compose:

```
        environment:
          BASE_URL: /gaps
```

But all I get is this:

![image](https://user-images.githubusercontent.com/39491297/114105200-fb603c00-9891-11eb-9fbc-ad51a820eeb1.png)

**Comment by JasonHHouse** [comment]
It's a pretty big change and I've been working on it for a while. It's pretty close to done. Shouldn't be too much longer.

**Comment by JasonHHouse** [comment]
I have a working build with no base URL and with a base URL, I'm using 'gaps' as the test URL. This passed tests with the URL enabled and off. Unraid is updated with an input to set the URL. It can also just be set as a Docker property. `BASE_URL`

**Comment by BEisem** [comment]
Commenting because I'm really interested in this as well. Hope to get a version to test soon!

**Comment by JasonHHouse** [comment]
The Docker build is published so you should be able to test using the BASE_URL environment variable. Let me know how it goes, @BEisem.

**Comment by BEisem** [comment]
@JasonHHouse which tag? I'm using `housewrecker/gaps:latest` but I'm not seeing anything new. The last release was 5 days ago.

**Comment by JasonHHouse** [comment]
@BEisem `housewrecker/gaps:latest` That's when I released the update. Code isn't merged so this ticket is closed as I was waiting to make sure things worked correctly.

**Comment by BEisem** [comment]
Ok so I don't think it's working, unless I'm doing something wrong... I added `BASE_URL=/gaps` to the environment variable in my docker-compose file. When I go to domain.com/gaps I get the same thing shown here: https://github.com/JasonHHouse/gaps/issues/193#issuecomment-816279215

**Comment by JasonHHouse** [comment]
That means the change isn't working. I'll keep on it and let you know when I update.

**Comment by JasonHHouse** [comment]
I found the issue. I built and pushed the new docker container. Let me know how it goes.

**Comment by kzaoaai** [comment]
how to set this option for windows?
